src/frequenz/sdk/timeseries/_serialized_ringbuffer.py

```diff
@@ -0,0 +1,78 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with serialization support."""
+
+import pickle
+from datetime import datetime, timedelta
+from os.path import exists
+from typing import List, TypeVar
+
+import numpy as np
+import numpy.typing as npt
+
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+Container = TypeVar("Container", List[float], npt.NDArray[np.float64])
+
+VARS_TO_SERIALIZE = (
+    "_datetime_oldest",
+    "_datetime_newest",
+    "_buffer",
+    "_gaps",
+    "_sampling_period",
+    "_time_range",
+    "_time_index_alignment",
+)
+
+
+class SerializedRingBuffer(OrderedRingBuffer[Container]):
+    """Sorted ringbuffer with serialization support."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        buffer: Container,
+        sampling_period: timedelta,
+        path: str,
+        time_index_alignment: datetime = datetime(1, 1, 1),
+        load_from_disk: bool = True,
+    ) -> None:
+        """Initialize the time aware ringbuffer.
+
+        Args:
+            buffer: Instance of a buffer container to use internally.
+            sampling_period: Timedelta of the desired resampling period.
+            path: Path to where the data should be saved to and loaded from.
+            time_index_alignment: Arbitrary point in time used to align
+                timestamped data with the index position in the buffer.
+                Used to make the data stored in the buffer align with the
+                beginning and end of the buffer borders.
+                For example, if the `time_index_alignment` is set to
+                "0001-01-01 12:00:00", and the `sampling_period` is set to
+                1 hour and the length of the buffer is 24, then the data
+                stored in the buffer could correspond to the time range from
+                "2022-01-01 12:00:00" to "2022-01-02 12:00:00" (date chosen
+                arbitrarily here).
+            load_from_disk: Default True, if True, will load the data from disk
+                upon start.
+        """
+        super().__init__(buffer, sampling_period, time_index_alignment)
+        self._path = path
+
+        if load_from_disk and exists(self._path):
+            self.load()
+
+    def dump(self) -> None:
+        """Dump data to disk."""
+        with open(self._path, mode="wb+") as fileobj:
+            pickle.dump(
+                tuple(getattr(self, attr) for attr in VARS_TO_SERIALIZE),
+                fileobj,
+            )
```

Comment:
I think the correct way to control what to pickle is to use [`__getstate__()`](https://docs.python.org/3/library/pickle.html#object.__getstate__). Not a 100% sure, but I'm almost sure there was a magic method to control this.

Reply:
Ah good to know, I'll look into that